src/components/chips/js/chipsController.js

```diff
@@ -128,21 +215,25 @@
    */
   MdChipsCtrl.prototype.getChipBuffer = function() {
     if (this.mdAutocompleteCtrl) {
-      this.$log.error('md-autocomplete not yet supported');
-    } else if (this.hasInputElement) {
-      this.$log.error('user-provided inputs not yet supported');
-    } else {
-      return this.chipBuffer;
+      throw Error('getChipBuffer should not be called if there is an md-autocomplete');
     }
+    return !this.userInputElement ? this.chipBuffer :
+        this.userInputNgModelCtrl ? this.userInputNgModelCtrl.$viewValue :
+            this.userInputElement[0].value;
   };
 
 
   /**
-   * Resets the input buffer.
+   * Resets the input buffer be it the internal input or user provided input element.
```

Comment:
`Resets the input buffer for either the internal input or the user provided input element.`